### PR TITLE
tidb-binlog: support binlog for clustered index

### PR DESCRIPTION
**Pull request by lysu**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions) **version-specific**. See https://github.com/pingcap/docs-cn/pull/6405#discussion_r644942446.
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

https://github.com/pingcap/tidb/issues/24133
https://github.com/pingcap/tidb/pull/24135
https://github.com/pingcap/tidb-binlog/pull/1040

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [x] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-binlog/tidb-binlog-overview.md [hunk 1]
```diff
@@ -49,9 +49,14 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
 
 ## 注意事项
 
+* TIDB Binlog 和 TiDB 在 v5.1 版本中解决了 v5.0 版本中引入的聚簇索引与 TiDB Binlog 不兼容问题。在升级 TiDBlog 和 TiDB Server 到 v5.1 版本后：开启 TiDB Binlog 后 TiDB 支持创建聚簇索引表；聚簇索引表的数据插入、删除和更新动作支持通过 TiDB Binlog 同步到下游。对于聚簇索引表同步需注意:
+
+    - 如果从 v5.0 版本手动控制组件升级顺序进行升级，请确保将 TiDB Binlog 升级至 v5.1 版本后再将 TiDB Server 升级至 v5.1 版本。
+    - 推荐配置上下游 TiDB 系统变量 [`tidb_enable_clustered_index`](/system-variables.md#tidb_enable_clustered_index-从-v50-版本开始引入)为一致的值来保证上游下游 TiDB 聚簇索引表结构一致。
+
```
Comment (lysu) [inline tidb-binlog/tidb-binlog-overview.md:56]: this part can be delete when cherry-picked to 5.0, but others can leave in 5.0 to advise user upgrade to 5.1 if they want to use binlog with clustered index

**Comment by qiancai** [comment]
@lysu Since this feature is for v5.1, would you please let us know that why 5.0 is also selected in the PR description?

**Comment by lysu** [comment]
> @lysu Since this feature is for v5.1, would you please let us know that why 5.0 is also selected in the PR description?

because this feature just fixes a leak function in v5.0, maybe it's better to modify v5.0 too, to let the user see v5.0 can decide whether choose v5.1?

we just need to add upgrade advice in v5.0 like https://github.com/pingcap/docs-cn/pull/6405#discussion_r644942446

File: tidb-binlog/tidb-binlog-overview.md [hunk 2]
```diff
@@ -49,9 +49,14 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
 
 ## 注意事项
 
+* TIDB Binlog 和 TiDB 在 v5.1 版本中解决了 v5.0 版本中引入的聚簇索引与 TiDB Binlog 不兼容问题。在升级 TiDBlog 和 TiDB Server 到 v5.1 版本后：开启 TiDB Binlog 后 TiDB 支持创建聚簇索引表；聚簇索引表的数据插入、删除和更新动作支持通过 TiDB Binlog 同步到下游。对于聚簇索引表同步需注意:
```
Comment (cfzjywxk) [inline tidb-binlog/tidb-binlog-overview.md:52]: Do we need to add the upgration guide document together with this or later?

**Review by cfzjywxk — COMMENTED** [review]
/lgtm

File: clustered-indexes.md [hunk 3]
```diff
@@ -141,11 +141,12 @@ mysql> SELECT TIDB_PK_TYPE FROM information_schema.tables WHERE table_schema = '
 目前 TiDB 的聚簇索引具有以下两类限制：
 
 - 明确不支持且没有支持计划的使用限制：
-    - 不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后 TiDB 只允许创建单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请使用 [TiCDC](/ticdc/ticdc-overview.md)。
     - 不支持与 [`SHARD_ROW_ID_BITS`](/shard-row-id-bits.md) 一起使用；[`PRE_SPLIT_REGIONS`](/sql-statements/sql-statement-split-region.md#pre_split_regions) 在聚簇索引表上不生效。
     - 不支持对聚簇索引表进行降级。如需降级，请使用逻辑备份工具迁移数据。
 - 另一类是尚未支持，但未来有计划支持的使用限制：
     - 尚未支持通过 `ALTER TABLE` 语句增加、删除、修改聚簇索引。
+- 特定版本的限制    
+    - v5.0 版本不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后 TiDB 只允许创建单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请升级至 v5.1 版本或使用 [TiCDC](/ticdc/ticdc-overview.md)。
```
Comment (TomShawn) [inline clustered-indexes.md:149]:
```suggestion
    - 在 v5.0 版本中，聚簇索引不支持与 TiDB Binlog 一起使用。开启 TiDB Binlog 后，TiDB 只允许创建单个整数列作为主键的聚簇索引；已创建的聚簇索引表的数据插入、删除和更新动作不会通过 TiDB Binlog 同步到下游。如需同步聚簇索引表，请升级至 v5.1 版本或使用 [TiCDC](/ticdc/ticdc-overview.md)。
```

File: tidb-binlog/tidb-binlog-overview.md (hunk 2 above)
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:52]:
```suggestion
* TiDB Binlog 和 TiDB 在 v5.1 版本中解决了 v5.0 版本中引入的聚簇索引与 TiDB Binlog 不兼容问题。在升级 TiDB Binlog 和 TiDB Server 到 v5.1 版本后：开启 TiDB Binlog 后，TiDB 支持创建聚簇索引表；聚簇索引表的数据插入、删除和更新操作支持通过 TiDB Binlog 同步到下游。对于同步聚簇索引表时需注意：
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 4]
```diff
@@ -49,9 +49,14 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
...
+
+    - 如果从 v5.0 版本手动控制组件升级顺序进行升级，请确保将 TiDB Binlog 升级至 v5.1 版本后再将 TiDB Server 升级至 v5.1 版本。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:54]:
```suggestion
    - 如果从 v5.0 版本手动控制组件升级顺序进行升级，请确保先将 TiDB Binlog 升级至 v5.1 版本后再将 TiDB Server 升级至 v5.1 版本。
```

File: tidb-binlog/tidb-binlog-overview.md [hunk 5]
```diff
@@ -49,9 +49,14 @@ TiDB Binlog 集群主要分为 Pump 和 Drainer 两个组件，以及 binlogctl
...
+    - 推荐配置上下游 TiDB 系统变量 [`tidb_enable_clustered_index`](/system-variables.md#tidb_enable_clustered_index-从-v50-版本开始引入)为一致的值来保证上游下游 TiDB 聚簇索引表结构一致。
```
Comment (TomShawn) [inline tidb-binlog/tidb-binlog-overview.md:55]:
```suggestion
    - 推荐将上下游的 TiDB 系统变量 [`tidb_enable_clustered_index`](/system-variables.md#tidb_enable_clustered_index-从-v50-版本开始引入) 配置为一致的值来保证上下游 TiDB 聚簇索引表结构一致。
```

**Review by TomShawn — COMMENTED** [review]
Some minor comments

**Review by TomShawn — APPROVED** [review]
LGTM

**Pull request merged**